examples/stress_tests/many_materials.rs

```diff
@@ -0,0 +1,121 @@
+//! Benchmark to test rendering many animated materials
+
+use bevy::{
+    diagnostic::{FrameTimeDiagnosticsPlugin, LogDiagnosticsPlugin},
+    prelude::*,
+    utils::HashSet,
+    window::{PresentMode, WindowPlugin, WindowResolution},
+};
+use std::f32::consts::PI;
+
+#[derive(Component)]
+struct Floor;
+
+fn main() {
+    App::new()
+        .add_plugins((
+            DefaultPlugins.set(WindowPlugin {
+                primary_window: Some(Window {
+                    resolution: WindowResolution::new(1920.0, 1080.0)
+                        .with_scale_factor_override(1.0),
+                    title: "many_materials".into(),
+                    present_mode: PresentMode::AutoNoVsync,
+                    ..default()
+                }),
+                ..default()
+            }),
+            FrameTimeDiagnosticsPlugin,
+            LogDiagnosticsPlugin::default(),
+        ))
+        .add_systems(Startup, setup)
+        .add_systems(Update, (animate_materials, make_materials_unique))
+        .run();
+}
+
+fn setup(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    let n = 4;
```

Comment:
Since this is a stress test this should probably be configurable using argh. Something like bevymark.

Reply:
Maybe giving it a specific total would be more useful though.

Reply:
If you're making it configurable with argh, please copy the cfg blocks from the other stress tests using it (eg. many_cubes) as `argh::from_env()` panics in wasm.